buildSrc/src/main/java/org/elasticsearch/gradle/plugin/PluginPropertiesExtension.java

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.plugin;
+
+import org.gradle.api.Project;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A container for plugin properties that will be written to the plugin descriptor, for easy
+ * manipulation in the gradle DSL.
+ */
+public class PluginPropertiesExtension {
+    private String name;
+
+    private String version;
+
+    private String description;
+
+    private String classname;
+
+    /** Other plugins this plugin extends through SPI */
+    private List<String> extendedPlugins = new ArrayList<>();
+
+    private boolean hasNativeController;
+
+    /** Indicates whether the plugin jar should be made available for the transport client. */
+    private boolean hasClientJar;
+
+    /** True if the plugin requires the elasticsearch keystore to exist, false otherwise. */
+    private boolean requiresKeystore;
+
+    /** A license file that should be included in the built plugin zip. */
+    private File licenseFile;
+
+    /**
+     * A notice file that should be included in the built plugin zip. This will be
+     * extended with notices from the {@code licenses/} directory.
+     */
+    private File noticeFile;
+
+    private Project project;
+
+    public PluginPropertiesExtension(Project project) {
+        Object version = project.getVersion();
+
+        this.name = project.getName();
+        this.version = (version != null && version instanceof String)? String.valueOf(version) : null;
+        this.project = project;
+    }
+
+    @Input
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Input
+    public String getVersion() {
+        return version;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    @Input
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    @Input
+    public String getClassname() {
+        return classname;
+    }
+
+    public void setClassname(String classname) {
+        this.classname = classname;
+    }
+
+    @Input
+    public List<String> getExtendedPlugins() {
+        return extendedPlugins;
+    }
+
+    public void setExtendedPlugins(List<String> extendedPlugins) {
+        this.extendedPlugins = extendedPlugins;
```

Comment:
We should add a null check on the arguments.

Reply:
If we marked it `final` as per above comment then this setter goes away. No?

Reply:
I was considering making it final, going away with the setter and having an `extendedPlugin` method to add them, but I didn't want to impose that on this PR, the comment on final did sneak in trough, so sorry for the confusion. 
If we don't do the above, we'll need to keep the setter and the field can't be final, so just add the null check and we are fine.

Reply:
No worries. Just to make sure I am not confused. I will remote the `final` keyword from the `extendedPlugins` array list field and add a `setter` with a null check to only set the `extendedPlugins` if its not null. This way it will either have extended plugins or be empty as per declaration 👍